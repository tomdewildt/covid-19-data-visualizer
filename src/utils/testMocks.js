/* eslint-disable react/prop-types */
import React from "react";

// Components
jest.mock( "@deck.gl/react", () => ( { children } ) => (
    <mock-deckgl>{ children }</mock-deckgl>
) );

// Assets
jest.mock( "../assets/gemeenten.json", () => ( [
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 1", Gemeentecode: 1, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 2", Gemeentecode: 2, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
    {
        type: "Feature",
        properties: { Gemeente: "Gemeente 3", Gemeentecode: 3, Coordinaten: [] },
        geometry: { type: "MultiPolygon", coordinates: [] },
    },
] ) );

