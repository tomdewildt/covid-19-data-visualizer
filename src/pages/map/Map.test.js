import React from "react";
import { axe } from "jest-axe";

import { renderWithTheme } from "../../utils/testUtils";

import PageMap from "./Map";

describe( "Map", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithTheme( <PageMap /> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithTheme( <PageMap /> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );
} );
