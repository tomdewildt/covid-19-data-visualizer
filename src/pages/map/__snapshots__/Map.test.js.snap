// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders consistently 1`] = `
<DocumentFragment>
  <main
    class=""
    id="map"
  />
</DocumentFragment>
`;
